storage/archive/ha_archive.cc

```diff
@@ -1771,7 +1777,7 @@ int ha_archive::info(uint flag)
   if (flag & HA_STATUS_AUTO)
   {
     if (init_archive_reader())
-      DBUG_RETURN(errno);
+      DBUG_RETURN(HA_ERR_CRASHED_ON_USAGE);
```

Comment:
Please don't change errno to HA_ERR_CRASHED_ON_USAGE.
This will hide the original error !